### PR TITLE
feat: subreddit card

### LINKED ISSUE #139: Support for reddit
**Is your feature request related to a problem? Please describe.**
I use namae.dev a lot for trying new names for my new project, and I thing it would be useful if it also searches in reddit

**Describe the solution you'd like**
Add a information checking if r/\<project name\> exists

### PR DESCRIPTION
**Pull request by ArnaudLier**
fixes #139

### CODE REVIEW

**Comment by uetchy** [comment]
Wow, you are so fast! I will merge this as soon as all the errors are resolved.
As lgtm bot said, you are missing `export` in the Subreddit.tsx

**Comment by ArnaudLier** [comment]
> Wow, you are so fast! I will merge this as soon as all the errors are resolved. As lgtm bot said, you are missing `export` in the Subreddit.tsx
Hmm, I resolved this error already, the Vercel deployment failed but I can't access the project on Vercel so can't see what's wrong?

**Comment by uetchy** [comment]
k fixed misc type errors that weren't related to this PR. Run `tsc --noEmit` and fix all the errors and ping me again.

File: public/locales/en/translation.json [hunk 1]
```diff
@@ -41,6 +41,7 @@
     "s3": "AWS S3",
     "slack": "Slack",
     "spectrum": "Spectrum",
+    "subreddit": "Subreddit",
```
Comment (uetchy) [inline public/locales/en/translation.json:44]: Fix the translation key (other languages as well) to `reddit` as it is referenced as `t('providers.reddit')`.

**Review by uetchy — CHANGES_REQUESTED** [review]

File: src/components/cards/providers/Subreddit.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { FaReddit } from 'react-icons/fa';
+
+import { Card, Repeater, DedicatedAvailability } from '../core';
+
+const SubredditCard: React.FC<{ query: string }> = ({ query }) => {
+  const { t } = useTranslation();
+  const normalizedQuery = normalize(query, {
```
Comment (uetchy) [inline src/components/cards/providers/Subreddit.tsx:9]: import `normalize`
Also always run `tsc --noEmit` before committing to make sure there are no other errors.

Comment (ArnaudLier) [reply src/components/cards/providers/Subreddit.tsx:10]: Wow, I'm really tired :sweat_smile:

**Comment by uetchy** [comment]
I'd say congrats, but there are some issues.

Screen Shot 2022-03-10 at 3 00 05

An HTTP request to the `/api/services/reddit/...` was failed because the folder was named `subreddit`. Rename `api/services/subreddit` to `api/services/reddit` to make it work.

Also specify `message` at `<DedicatedAvailability />` to avoid wacky empty tooltips.

---

Looking at Reddit's subreddit naming policy, they stated `_` is the only special character they accept other than alphanumerics.

Screen Shot 2022-03-10 at 3 14 00

Currently `normalize` is called with `allowUnderscore` option set `false` which prevents the use of underscore. Fix this setting `true`, and the second item of `moreNames` in the `[query].ts` as well.

```
const moreNames = [
  `get${lowerCase}`,
  `${lowerCase}-team`,
];
```

**Review by uetchy — APPROVED** [review]

**Comment by uetchy** [comment]
All look fine👍

**Pull request merged**